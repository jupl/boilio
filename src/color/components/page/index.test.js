import {props, actions} from '.'

describe('<Color.ColorPage>', () => {
  it('should inject props from store as expected', () => {
    const store = {
      color: 'red',
      other: 'else',
    }
    expect(props(store)).toMatchSnapshot()
  })

  it('should inject action props as expected', () => {
    const dispatch = jest.fn()
    const {onPreviousColor, onNextColor} = actions(dispatch)
    onPreviousColor()
    onNextColor()
    expect(dispatch.mock.calls).toMatchSnapshot()
  })
})
