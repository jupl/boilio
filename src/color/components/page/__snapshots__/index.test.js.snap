exports[`<Color.ColorPage> should inject action props as expected 1`] = `
Array [
  Array [
    Object {
      "type": "COLORS_PREVIOUS_COLOR",
    },
  ],
  Array [
    Object {
      "type": "COLORS_NEXT_COLOR",
    },
  ],
]
`;

exports[`<Color.ColorPage> should inject props from store as expected 1`] = `
Object {
  "color": "red",
  "style": undefined,
}
`;
